src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/service/ScottishBankruptOfficerSearchServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+        searchFilters.setForename1(FORENAME);
+        searchFilters.setSurname(SURNAME);
+        searchFilters.setDateOfBirth(DATE_OF_BIRTH);
+        searchFilters.setPostcode(POSTCODE);
+
+        ScottishBankruptOfficerSearch search = new ScottishBankruptOfficerSearch();
+        search.setStartIndex(START_INDEX);
+        search.setItemsPerPage(ITEMS_PER_PAGE);
+        search.setFilters(searchFilters);
+
+        ScottishBankruptOfficerSearchEntity searchEntity = new ScottishBankruptOfficerSearchEntity();
+        ScottishBankruptOfficerSearchResultsEntity resultsEntity = new ScottishBankruptOfficerSearchResultsEntity();
+        ScottishBankruptOfficerSearchResults expectedSearchResults = new ScottishBankruptOfficerSearchResults();
+
+        when(transformer.convertToSearchEntity(search)).thenReturn(searchEntity);
+        when(dao.getScottishBankruptOfficers(searchEntity)).thenReturn(resultsEntity);
+        when(transformer.convertToSearchResults(resultsEntity)).thenReturn(expectedSearchResults);
+
+        ScottishBankruptOfficerSearchResults searchResults = service.searchScottishBankruptOfficers(search);
+        assertEquals(searchResults, expectedSearchResults);
+    }
+
+    @Test
+    @DisplayName("No results returned when retrieving Scottish bankrupt officers")
+    void testNoResultsReturnsForScottishBankruptOfficersSearch(){
+        ScottishBankruptOfficerSearchEntity mockTransformerResponse = new ScottishBankruptOfficerSearchEntity();
+        when(transformer.convertToSearchEntity(any(ScottishBankruptOfficerSearch.class))).thenReturn(mockTransformerResponse);
+        when(dao.getScottishBankruptOfficers(any(ScottishBankruptOfficerSearchEntity.class))).thenReturn(null);
+
+        ScottishBankruptOfficerSearch search = new ScottishBankruptOfficerSearch();
+        ScottishBankruptOfficerSearchResults searchResults = service.searchScottishBankruptOfficers(search);
+
+        assertNull(searchResults);
+    }
+
+    @Test
+    @DisplayName("Search for bankrupt officer by id")
+    void testScottishBankruptSearchByID() {
+        ScottishBankruptOfficerDetailsEntity mockRepoResponse = new ScottishBankruptOfficerDetailsEntity() {{
+            setEphemeralKey(EPHEMERAL_KEY);
+        }};
+        when(dao.getScottishBankruptOfficer(eq(EPHEMERAL_KEY))).thenReturn(mockRepoResponse);
+        ScottishBankruptOfficerDetails mockTransformerResponse = new ScottishBankruptOfficerDetails() {{
+            setEphemeralKey(EPHEMERAL_KEY);
+
+        }};
+        when(transformer.convertToDetails(mockRepoResponse)).thenReturn(mockTransformerResponse);
+
+        ScottishBankruptOfficerDetails result = service.getScottishBankruptOfficer(EPHEMERAL_KEY);
+        assertEquals(EPHEMERAL_KEY, result.getEphemeralKey());
+    }
+
+    @Test
+    @DisplayName("No officer found with get by ID")
+    void testNoOfficerFoundByID(){
+        when(dao.getScottishBankruptOfficer(ArgumentMatchers.anyString())).thenReturn(null);
```

Comment:
This should be `when(dao.getScottishBankruptOfficer(EPHEMERAL_KEY)).thenReturn(null);`

Reply:
Changed